cpp/include/cudf/detail/iterator.cuh

```diff
@@ -153,6 +228,53 @@ auto inline make_validity_iterator(column_device_view const& column)
       validity_accessor{column});
 }
 
+/**
+ * @brief Constructs a constant iterator over a scalar's value.
+ *
+ * Dereferencing the returned iterator returns a `Element`.
+ * 
+ * For `p = *(iter + i)`, `p` is the value stored in the scalar.
+ *
+ * @throws `cudf::logic_error` if scalar datatype and Element type mismatch.
+ * @throws `cudf::logic_error` if scalar is null.
+ *
+ * @tparam Element The type of elements in the scalar
+ * @param scalar_value The scalar to iterate
+ * @return auto Iterator that returns scalar value
+ */
+template <typename Element>
+auto inline make_scalar_iterator(scalar const& scalar_value)
+{
+  CUDF_EXPECTS(data_type(experimental::type_to_id<Element>()) == scalar_value.type(), "the data type mismatch");
+  CUDF_EXPECTS(scalar_value.is_valid(), "the scalar value must be valid");
+  using ScalarType = experimental::scalar_type_t<Element>;
+  return thrust::make_constant_iterator(static_cast<ScalarType const*>(&scalar_value)->value());
+}
+
+/**
+ * @brief Constructs a constant pair iterator over a scalar's value and its validity.
+ *
+ * Dereferencing the returned iterator returns a `thrust::pair<Element, bool>`.
+ *  
+ * If scalar is valid, then for `p = *(iter + i)`, `p.first` contains 
+ * the value of the scalar and `p.second == true`.
+ *
+ * Else, if the scalar is null, then the value of `p.first` is undefined and `p.second == false`. 
+ * 
+ * @throws `cudf::logic_error` if scalar datatype and Element type mismatch.
+ *
+ * @tparam Element The type of elements in the scalar
+ * @param scalar_value The scalar to iterate
+ * @return auto Iterator that returns scalar, and validity of the scalar in a pair
+ */
+template <typename Element>
```

Comment:
While this may seem odd, it would be convenient for `make_pair_iterator` to be uniform between a `scalar` and a `column`. Adding this template param is necessary for that. 
```suggestion
template <typename Element, bool is_null>
```

Reply:
since column_device_view has `pair_begin` and `pair_end` now, is it okay to remove this suggestion for uniformity and also remove `make_pair_iterator(column_device_view)`?

Reply:
removed.